lib/src/app.dart

```diff
@@ -106,6 +112,56 @@ class _AppState extends ConsumerState<Application> {
       setOptimalDisplayMode();
     }
 
+    ref.read(localNotificationServiceProvider).init();
+
+    ref.listenManual(challengesProvider, (prev, current) {
+      final prevIds = prev!.value!.inward.map((challenge) => challenge.id);
```

Comment:
I don't think we can assume the `prev` and the `value` are never null. I'd write that if you want prevIds to be defined no matter what:
```dart
final prevIds = prev?.value?.inward.map((challenge) => challenge.id) ?? [];
```

Reply:
i think we just want to exit if prev is null though.
if prev is null it means that we have the initial state of the provider, or the app has just opened, and if there is an incoming challenge request when we open the app for the first time, then immediately after opening the app the user will get a notification for each challenge request they currently have.
i think that would be pretty annoying, and if they are opening the app for the first time they will just see that they have challenge requests based on the notification pip on the challenge button.

Reply:
maybe i missunderstand how the ! works here, i assumed it just exited on a null value, does it also throw an error?

Reply:
i misunderstood how the ! operator was used, i thought it just exited i didnt realise it also threw an error.